addons/sale_mrp/tests/test_sale_mrp_lead_time.py

```diff
@@ -0,0 +1,142 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import timedelta
+
+from odoo import fields
+from odoo.tests import common
+
+
+class TestSaleMrpLeadTime(common.TransactionCase):
+
+    def setUp(self):
+        super(TestSaleMrpLeadTime, self).setUp()
+
+        # Useful models
+        self.Product = self.env['product.product']
+        self.ProcurementOrder = self.env['procurement.order']
+        self.SaleOrder = self.env['sale.order']
+
+        self.company = self.env.ref('base.main_company')
+        self.partner = self.env.ref('base.res_partner_1')
+        self.price_list = self.env.ref('product.list0')
+        self.uom_unit = self.env.ref('product.product_uom_unit')
+        self.warehouse = self.env.ref('stock.warehouse0')
+
+        route_manufacture = self.warehouse.manufacture_pull_id.route_id.id
+        route_mto = self.warehouse.mto_pull_id.route_id.id
+
+        # Create product A with Manufacturing Lead Time and Customer Lead Time
+        self.productA = self.Product.create({
```

Comment:
don't get me wrong, I am glad you are at least writing tests and this isn't
specific to this one, but why only ever test obvious cases?

What if sale order date in past or future? What if days is negative? What
if days is less than 1.  What if days is 0. What if days is 3.25.
Those kind of cases not only pick up regressions but prevent them by
explicitly documenting expected behaviour.

On Fri, 13 May 2016 9:48 PM Thibault Delavallée <notifications@github.com>
wrote:

> In addons/sale_mrp/tests/test_sale_mrp_lead_time.py
> <https://github.com/odoo/odoo/pull/11685#discussion_r63158527>:
>
> > +        # Useful models
> > +        self.Product = self.env['product.product']
> > +        self.ProcurementOrder = self.env['procurement.order']
> > +        self.SaleOrder = self.env['sale.order']
> > +
> > +        self.company = self.env.ref('base.main_company')
> > +        self.partner = self.env.ref('base.res_partner_1')
> > +        self.price_list = self.env.ref('product.list0')
> > +        self.uom_unit = self.env.ref('product.product_uom_unit')
> > +        self.warehouse = self.env.ref('stock.warehouse0')
> > +
> > +        route_manufacture = self.warehouse.manufacture_pull_id.route_id.id
> > +        route_mto = self.warehouse.mto_pull_id.route_id.id
> > +
> > +        # Create product A with Manufacturing Lead Time and Customer Lead Time
> > +        self.productA = self.Product.create({
>
> See above.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/odoo/odoo/pull/11685/files/5f6f2261baacda4671fcc74e964e619121e0236f#r63158527>
>